lib/open_project/translations/models/locales_updater_configuration.rb

```diff
@@ -1,14 +1,46 @@
 require 'yaml'
 
 class LocalesUpdaterConfiguration
-  CONFIGURATION_FILE = ENV['CONFIGURATION_FILE'] || begin
-    # root of OpenProject-Translation
-    configuration_file = Pathname(__FILE__) + '../../../../../'
-    configuration_file + 'configuration.yml'
+  REQUIRED_KEYS = [:uri, :api_key, :project_id, :version,
+                   :version, :branch]
+
+  def self.path_to_configuration_file
+    ENV['CONFIGURATION_FILE'] || begin
+      # root of OpenProject-Translation
+      configuration_file = Pathname(__FILE__) + '../../../../../'
+      configuration_file + 'configuration.yml'
+    end
   end
 
   def self.configuration
-    # todo raise error if something is missing
-    @configuration ||= YAML.load_file(CONFIGURATION_FILE)
+    @configuration ||= YAML.load_file(path_to_configuration_file)
+    check_for_structure
+    check_for_required_keys
+    @configuration
+  end
+
+  def self.check_for_structure
+    unless @configuration.has_key?(:plugins)
+      raise "Configuration file doesn't have :plugins key"
+    end
+    if @configuration[:plugins].nil?
+      raise "Configuration file has empty :plugins key"
+    end
+  end
+
+  def self.check_for_required_keys
+    @configuration[:plugins].each do |plugin_name, specifics|
```

Comment:
Unused block argument - `specifics`. If it's necessary, use `_` or `_specifics` as an argument name to indicate that it won't be used.